sql/udafs.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: User-defined Aggregate Functions (UDAFs)
+description: 'Implement custom aggregation logic in Arroyo SQL with UDAFs'
+---
+
+Arroyo SQL supports User-Defined Aggregate Functions, allowing you to extend the system's capabilities with custom aggregation logic written in Rust. 
+UDAFs can be registered via the SQL API call or the Web UI.
+
+## How to Define a UDAF
+
+In Rust, UDAFs are defined as functions that take some number of `Vec<T>` as an argument, where `T` is any valid [SQL data type](/sql/data-types). 
+
+### Example
+
+Here's a simple UDAF for calculating the median of a set of integers:
+
+```rust
+fn my_median(args: Vec<i64>) -> Option<f64> {
+    if args.is_empty() {
+        return None;
+    }
+    let mut args = args;
```

Comment:
future work, but we should probably support this in the parameter list

Reply:
Was curious how this fails, turns out it doesn't! Gonna update the example to the more natural function.